### DISCUSSION TITLE
HTTP/2 always enabled with fixed upload

### DISCUSSION THREAD

**Discussion by psychedelicu** [ZoeyVid/NPMplus#672] [Q&A] [answered]
What exactly this setting does? "HTTP/2 always enabled with fixed upload"

I'm asking because i use NPM plus with cloudflare and nextcloud aio on unraid and my upload/download speeds on nextcloud is very low, somewhere between 5-10mb, and i have a gigabyte connection.

**Comment by Zoey2936** [comment]
try to disable cloudflare, cloudflare isn't made to proxy file downloads

**Reply by psychedelicu** [reply]
I know, the CNAME record for nextcloud is unproxied. But the speed is still low.

**Reply by Zoey2936** [reply]
so fully disabled?

**Reply by Zoey2936** [reply]
since in my tests NPMplus has upload rates higher then 50 mb/s and then the network downlink limits

**Comment by Zoey2936** [comment]
> What exactly this setting does? "HTTP/2 always enabled with fixed upload"

http2 is always set to "on" and I've added https://github.com/ZoeyVid/NPMplus/blob/b4b4ff8708e557b5538244129054d41b352d1473/rootfs/usr/local/nginx/conf/nginx.conf#L36 because of: https://blog.cloudflare.com/delivering-http-2-upload-speed-improvements

**Comment by Zoey2936** [comment, marked as answer]
see: https://github.com/nextcloud/all-in-one/discussions/4307#discussioncomment-8727883
